homeassistant/components/tplink/entity.py

```diff
@@ -240,33 +227,72 @@ def available(self) -> bool:
         return self.coordinator.last_update_success and self._attr_available
 
 
+def _category_for_feature(feature: Feature | None) -> EntityCategory | None:
```

Comment:
Moved out of the class so can be called by other platforms

Reply:
We should not do this. It should be solely in the responsibility of the base entity class to decide on this to keep things simple.

Reply:
Now in `descriptions.py`